tests/framework/artifact/test_hdf.py

```diff
@@ -272,9 +272,12 @@ def test_EntityKey_init_failure():
     bad_keys = ['hello', 'a.b.c.d', '', '.', '.coconut', 'a.', 'a..c']
 
     for k in bad_keys:
-        with pytest.raises(ValueError):
+        with pytest.raises(ValueError) as err_info:
```

Comment:
You can use the `match` keyword argument for `pytest.raises` here instead of making a separate assert.  I'd link you to the docs, but I can't google stuff with my current internet.